### PR TITLE
Databasejoin

### PR DESCRIPTION
**Pull request by joshFeenberg**
It should have everything needed for the API calls as far as the database is concerned.

### CODE REVIEW

File: .idea/.gitignore
Comment (srames50) [inline .idea/.gitignore:1]: make sure we ignore all files that need to be ignored, reach out if you have questions

File: .idea/.name
Comment (srames50) [inline .idea/.name:1]: is this one of the files you were talking about that is randomly created

File: backend/demo/src/main/java/com/example/demo/Controller.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@RestController
+@RequestMapping("/api")
+public class Controller {
+    
+    @GetMapping("/hello")
+    public String sayHello() {
+        return "Hello World!"; 
+    }
+  
+
```
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/Controller.java]: If the code commented is not necessary, we should remove it when pushing to main

File: backend/demo/src/main/java/com/example/demo/Controller.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                        1,
+                        "Sample Event",
+                        "This is a test event to demonstrate the Event class.",
+                        LocalDateTime.of(2024, 5, 22, 10, 0),
+                        LocalDateTime.of(2024, 5, 22, 12, 0),
+                        LocalDateTime.now(),
+                        null
+                ));
+    }*/
+
```
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/Controller.java:28]: Should probably remove dummy data when pushing to main

File: backend/demo/src/main/java/com/example/demo/Event/EventConfig.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Configuration
+public class EventConfig {
+    @Bean
+    CommandLineRunner EventcommandLineRunner(EventRepository eventRepository) {
```
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/Event/EventConfig.java:25]: Add comments to specify what is happening in this file

File: backend/demo/src/main/java/com/example/demo/Event/EventController.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.example.demo.Event;
+
+
```
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/Event/EventController.java]: general description for what the file is

File: backend/demo/src/main/java/com/example/demo/EventStudent/EventStudent.java
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/EventStudent/EventStudent.java:1]: Comment what the file is being used for

File: backend/demo/src/main/java/com/example/demo/student/Sservice.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.example.demo.student;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class Sservice {
```
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/student/Sservice.java:13]: specify what sservice is

File: backend/demo/src/main/java/com/example/demo/student/StudentConfig.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+
+@Configuration
+public class StudentConfig {
+    @Bean
+    CommandLineRunner commandLineRunner(StudentRepo studentRepository) {
+        return args -> {
```
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/student/StudentConfig.java:21]: do we need this test code in main?

File: backend/demo/src/main/java/com/example/demo/student/StudentService.java
Comment (srames50) [inline backend/demo/src/main/java/com/example/demo/student/StudentService.java:1]: what is this file

File: frontend/lib/api_service.dart
Comment (srames50) [inline frontend/lib/api_service.dart:1]: say that this file is connecting frontend to backend and where you are getting API service from, and that http is getting called in main.dart

File: frontend/lib/main.dart
Comment (srames50) [inline frontend/lib/main.dart:1]: more comments for readability

File: backend/demo/src/main/java/com/example/demo/student/StudentConfig.java (hunk 6 above)
Comment (joshFeenberg) [reply backend/demo/src/main/java/com/example/demo/student/StudentConfig.java:21]: It's needed until we have real events and users

Comment (joshFeenberg) [reply backend/demo/src/main/java/com/example/demo/student/StudentConfig.java:21]: I added functionality so we can send a post request now. We can talk about which is better over meeting.

**Review by VIkranthMandadi — COMMENTED** [review]
No additional comments other than Shyam's, thanks

**Pull request merged**